### PR TITLE
Master branch sync and Foxy fixup

### PR DESCRIPTION
**Pull request by henningkayser**

### CODE REVIEW

File: test_util.sh [hunk 1]
```diff
@@ -134,7 +134,7 @@ function run_test() {
 
 	if [ $result -ne $expected ] ; then
 		let "FAILED += 1"
-		test $expected -eq 0 && comment="Expected success, but failed ($result)." || comment="Expected failure ($expected), but succeeded"
+		test $expected -eq 0 && comment="Expected success, but failed ($result)." || comment="Expected failure ($expected), but succeeded ($result)"
```
Comment (rhaschke) [inline test_util.sh:137]: This is definitely not a change from master. When writing this, I was assuming that `$result` would be always zero in the else branch. However, we might expect failure with a different error id. So, while I accept this change, I ask you to file this for master first.
```suggestion
		test $expected -eq 0 && comment="Expected success, but failed ($result)." || comment="Expected failure ($expected), but got $result"
```

**Review by rhaschke — CHANGES_REQUESTED** [review]
I guess all of those changes are reasonable, but you are mixing different changes in a single huge commit. Please structure this PR into different, meaningful commits, e.g.:
on `master`:
- transition to clang-format-10
- update `test_utils.sh` (see below)

on `ros2`:
- rename branch master -> main
- update eloquent -> foxy
- merge master
- apply clang-format-10 changes to ros2-only files (this might be part of previous merge commit)

**Comment by henningkayser** [comment]
> I guess all of those changes are reasonable, but you are mixing different changes in a single huge commit. Please structure this PR into different, meaningful commits, e.g.:
> on `master`:
> 
> * transition to clang-format-10
> * update `test_utils.sh` (see below)

I agree about the `test_utils.sh`, I'm not sure I understand what you want to have merged into master regarding clang-format-10. Support for clang-format-10 was the reason why I did the sync.

> on `ros2`:
> 
> * rename branch master -> main
> * update eloquent -> foxy
> * merge master
> * apply clang-format-10 changes to ros2-only files (this might be part of previous merge commit)

Makes sense

File: test_util.sh (hunk 1 above)
Comment (henningkayser) [reply test_util.sh:137]: added in #120

**Comment by rhaschke** [comment]
IMO, this PR is still rather unclean: first you merge `master` (https://github.com/ros-planning/moveit_ci/pull/118/commits/5f5e5fc9745e718c520639aebf74d84205e2dbba) in a rather broken stage and then you file various fixup commits, most notably (https://github.com/ros-planning/moveit_ci/pull/118/commits/f758c97ab5e328aaa049e28299233c2d4ec8db45), which is doing much more than just _"Switching to Foxy"_. Many of the other fixup commits could be applied to the original `ros2` branch, before merging `master`.

I filed an alternative PR: #118, which also fixes a small issue you introduced with your merge: `catkin_tools` is not required for `ros2`, but `curl` is.

**Comment by henningkayser** [comment]
> IMO, this PR is still rather unclean: first you merge `master` ([5f5e5fc](https://github.com/ros-planning/moveit_ci/commit/5f5e5fc9745e718c520639aebf74d84205e2dbba)) in a rather broken stage and then you file various fixup commits, most notably ([f758c97](https://github.com/ros-planning/moveit_ci/commit/f758c97ab5e328aaa049e28299233c2d4ec8db45)), which is doing much more than just _"Switching to Foxy"_. Many of the other fixup commits could be applied to the original `ros2` branch, before merging `master`.
> 

I don't really see it that way. The merge really just pulls the latest changes from master first since it adds a lot of new stuff including unit tests, there is no reasonable way to fix those in advance. The commit ([f758c97](https://github.com/ros-planning/moveit_ci/commit/f758c97ab5e328aaa049e28299233c2d4ec8db45) is actually not doing much more than just _"Switching to Foxy"_, the "master->main" migration was done together with Foxy which is why it doesn't make any sense to split those.

> I filed an alternative PR: #118, which also fixes a small issue you introduced with your merge: `catkin_tools` is not required for `ros2`, but `curl` is.

Thanks for putting time into this.

**Comment by rhaschke** [comment]
> I don't really see it that way.

There were several merge conflicts that you resolved in one way and later fixed them via https://github.com/ros-planning/moveit_ci/commit/f758c97ab5e328aaa049e28299233c2d4ec8db45, e.g. this one:
https://github.com/ros-planning/moveit_ci/commit/f758c97ab5e328aaa049e28299233c2d4ec8db45#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L43-R47

IMO, a merge commit should fix all merge conflicts in a way that the commit itself is fine. However, let's look forward...

**Pull request closed without merging**